ADR/0045-guidelines-for-developing-tasks.md

```diff
@@ -0,0 +1,54 @@
+# 45. Guidelines for developing tasks
+
+* Date: 2024-11-06
+
+## Status
+
+Proposed
+
+## Context
+
+One of the architectural goals for Konflux is extensibility. The core system provides opinionated [build pipelines]
+and [release pipelines], but lets users extend those and/or create their own. Beyond this goal, however, there has
+not been any clear guidance about where contributions should be added or the types of functionality that are typical
+for the stages of software development.
+
+## Decision
```

Comment:
Would mentioning [Task](https://enterprisecontract.dev/docs/ec-policies/task_policy.html) [policies](https://enterprisecontract.dev/docs/ec-policies/build_task_policy.html) be of value here? As in: "Tasks contributed to Konflux need to adhere to the Build and Task policies"